packages/opentelemetry-exporter-collector-grpc/test/CollectorTraceExporter.test.ts

```diff
@@ -155,6 +155,15 @@ const testCollectorExporter = (params: TestParams) =>
         const args = spyLoggerWarn.args[0];
         assert.strictEqual(args[0], 'Headers cannot be set when using grpc');
       });
+      it('should warn about path in url', () => {
+        const spyLoggerWarn = sinon.stub(diag, 'warn');
+        collectorExporter = new CollectorTraceExporter({
+          serviceName: 'basic-service',
+          url: address + '/v1/trace',
+        });
+        const args = spyLoggerWarn.args[0];
+        assert.strictEqual(args[0], 'URL path cannot be set when using grpc');
```

Comment:
Your test is failing because `args` is undefined here. Looks like improper use of sinon most likely to me. Not sure what `spyLoggerWarn.args[0]` should return, but usually with sinon you see something like `spyLoggerWarn.firstCall().args[0]` to get the first argument of the first call.

Reply:
oh, I see the problem, `afterEach(() => { sinon.restore(); });` resets state between each test, but not between individual `describe`s I think.

Reply:
the `afterEach` callback gets called after every `it` test within the `describe` block that it is defined in.

Reply:
discussing offline...